app/src/main/java/com/blackboxteam/app/blackbox/util/UserSessionPersistence.java

```diff
@@ -0,0 +1,39 @@
+package com.blackboxteam.app.blackbox.util;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.widget.Toast;
+
+/**
+ * UserSessionPersistence
+ *
+ * Manages user session
+ */
+public class UserSessionPersistence {
+    private Context mContext;
+    private String user;
+
+    public UserSessionPersistence () {
+
+    }
+
+    // Save the user login/signIp info
+    public void saveUserInfo(String email, String password) {
+        SharedPreferences sharedPreferences;
+        sharedPreferences = mContext.getSharedPreferences("userInfo", Context.MODE_PRIVATE);
+
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString("username", email);
+        editor.putString("password", password);
+        editor.apply();
+    }
+
+    // get user information
+    public boolean getUserInfo(String sharedPrefName) {
+        SharedPreferences sharedPreferences;
```

Comment:
Instead of repeatedly getting shared preference, we could make it a dependency in the UserSessionPersistence Constructor